kotlin/android/app/src/main/java/dev/firezone/android/core/data/PreferenceRepositoryImpl.kt

```diff
@@ -48,8 +61,8 @@ internal class PreferenceRepositoryImpl
 
         override fun validateCsrfToken(value: String): Flow<Boolean> =
             flow {
-                val token = sharedPreferences.getString(CSRF_KEY, "")
-                emit(token == value)
+                val token = sharedPreferences.getString(CSRF_KEY, "").orEmpty()
+                emit(MessageDigest.isEqual(token.toByteArray(), value.toByteArray()))
```

Comment:
@pratikvelani There was a timing attack vulnerability here. `MessageDigest.isEqual` seems to be a secure compare instead.